sdl_android/src/main/java/com/smartdevicelink/api/screen/ScreenManager.java

```diff
@@ -0,0 +1,343 @@
+	 */
+	public void setTextField3Type(MetadataType textField3Type) {
+		this.textAndGraphicManager.setTextField3Type(textField3Type);
+	}
+
+	/**
+	 * Get the metadata type for textField3
+	 * @return a MetadataType value represents the metadata for textField3
+	 */
+	public MetadataType getTextField3Type() {
+		return this.textAndGraphicManager.getTextField3Type();
+	}
+
+	/**
+	 * Set the metadata type for the textField4
+	 * @param textField4Type a MetadataType value represents the metadata for textField4
+	 */
+	public void setTextField4Type(MetadataType textField4Type) {
+		this.textAndGraphicManager.setTextField4Type(textField4Type);
+	}
+
+	/**
+	 * Get the metadata type for textField4
+	 * @return a MetadataType value represents the metadata for textField4
+	 */
+	public MetadataType getTextField4Type() {
+		return this.textAndGraphicManager.getTextField4Type();
+	}
+
+	/**
+	 * Set softButtonObjects list and upload the images to the head unit
+	 * @param softButtonObjects the list of the SoftButtonObject values that should be displayed on the head unit
+	 */
+	public void setSoftButtonObjects(@NonNull List<SoftButtonObject> softButtonObjects) {
+		softButtonManager.setSoftButtonObjects(softButtonObjects);
+	}
+
+	/**
+	 * Get the soft button objects list
+	 * @return a List<SoftButtonObject>
+	 */
+	public List<SoftButtonObject> getSoftButtonObjects() {
+		return softButtonManager.getSoftButtonObjects();
+	}
+
+	/**
+	 * Get the SoftButtonObject that has the provided name
+	 * @param name a String value that represents the name
+	 * @return a SoftButtonObject
+	 */
+	public SoftButtonObject getSoftButtonObjectByName(@NonNull String name){
+		return softButtonManager.getSoftButtonObjectByName(name);
+	}
+
+	/**
+	 * Get the SoftButtonObject that has the provided buttonId
+	 * @param buttonId a int value that represents the id of the button
+	 * @return a SoftButtonObject
+	 */
+	public SoftButtonObject getSoftButtonObjectById(int buttonId){
+		return softButtonManager.getSoftButtonObjectById(buttonId);
+	}
+
+	public void setDisplayLayout(PredefinedLayout layout){
+		SetDisplayLayout setDisplayLayoutRequest = new SetDisplayLayout();
+		setDisplayLayoutRequest.setDisplayLayout(layout.toString());
+		internalInterface.sendRPCRequest(setDisplayLayoutRequest);
+	}
+
+	/**
+	 * Begin a multiple updates transaction. The updates will be applied when commit() is called<br>
+	 * Note: if we don't use beginTransaction & commit, every update will be sent individually.
+	 */
+	public void beginTransaction(){
+		softButtonManager.setBatchUpdates(true);
+		textAndGraphicManager.setBatchUpdates(true);
+	}
+
+	/**
+	 * Send the updates that were started after beginning the transaction
+	 * @param listener a CompletionListener that has a callback that will be called when the updates are finished
+	 */
+	public void commit(final CompletionListener listener){
+		allSubManagersFinishedUpdatingSuccesslly = true;
```

Comment:
Is there a reason this isn't a local variable? Just add `boolean allSubManagersFinishedUpdatingSuccesslly = true;` in the global scope of the new anonymous `CompletionListener` instantiation. I would also recommend changing the name as this is extremely verbose, perhaps `updateSuccessful`.